core/src/main/resources/hudson/tasks/Shell/config.groovy

```diff
@@ -27,3 +27,10 @@ f=namespace(lib.FormTagLib)
 f.entry(title:_("Command"),description:_("description",rootURL)) {
     f.textarea(name: "command", value: instance?.command, class: "fixed-width", 'codemirror-mode': 'shell', 'codemirror-config': "mode: 'text/x-sh'")
 }
+
+f.advanced() {
+    f.entry(title:_("Exit code to set build unstable"), field: "unstableReturn") {
+        f.number(clazz:"positive-number", name: "unstableReturn", value: instance?.unstableReturn, min:1, max:255, step:1)
```

Comment:
The is a general question if we should allow users to define exit code zero as an unstable build? I think better not.